### PR TITLE
Added ability to select a date range

### PR DESCRIPTION
**Pull request by aryaxt**
Note: because the delegate method for date range can have a null second date, I had to add nullability to the delegate method in order support Swift, and that required adding nullability to all methods and properties in the header file

Still doing some cleanup, will post a comment when it's ready

### CODE REVIEW

File: RSDayFlow/RSDFDatePickerView.h [hunk 1]
```diff
@@ -222,7 +255,19 @@
...
+ Tells the delegate that the user did select a date in RSDFSelectionModeRange.
+ 
+ The date picker view calls this method when the user successfully selects a date in the date picker view.
+ It does not call this method when you programmatically set the selection.
+ 
+ @param view The view whose date was selected.
+ @param startDate The selected start date for range.
+ @param endDate The selected end date for range.
+ */
+- (void)datePickerView:(RSDFDatePickerView * __nonnull)view didSelectStartDate:(NSDate * __nullable)startDate endDate:(NSDate * __nullable)endDate;
```
Comment (aryaxt) [inline RSDayFlow/RSDFDatePickerView.h]: Does startDate/endDate sound confusing since those names are also used for limiting start/end date?

**Comment by aryaxt** [comment]
@ruslanskorb This is good for review

**Comment by ruslanskorb** [comment]
@aryaxt Thanks!
I'll think about better names for dates of the selected range.
I would like to change the default look of the selected range to something like this:

selected_date_range

Can we do it?

**Comment by aryaxt** [comment]
Should be fine will look into it. Just wondering if it would look weird if the selection contains 2 months. Will give it a shot.

Build is failing because travis is running on an older version of Xcode.

**Comment by ruslanskorb** [comment]
@aryaxt I'll be glad to see it.
You don't need to worry about Travis, I know.

**Comment by aryaxt** [comment]
This is definitely trickier than I thought. Will look into it again this weekend. 
I think it would be nice to allow both styles, grouped-date-range and separated-date-range

**Comment by aryaxt** [comment]
@ruslanskorb Any thought on merging this the way it looks like and adding a second theme to have the date ranges grouped (similar to screen shot above) on a separate PR/version? 

My design requirement is the way I have it in the PR, and haven't had time to work on the grouped date range theme. Let me know so that I know whether I should wait for this to get merged or use my fork for range picking.

Thanks

**Comment by ruslanskorb** [comment]
Hi @aryaxt!
Unfortunately, at the moment I do not have time to thoroughly go through this PR. I want to merge it some later and release a new version with it and the grouped date ranges.
Thanks!

**Comment by aryaxt** [comment]
ok awesome, will keep this open and point to my branch till you review

**Comment by aryaxt** [comment]
Thanks for the awesome library
My new requirements are very specific to the app I'm building so won't make sense to add them here, will continue on my fork

**Pull request closed without merging**